hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/ContainerKeyMapperHelper.java

```diff
@@ -84,66 +124,108 @@ public static boolean reprocess(OMMetadataManager omMetadataManager,
+      
+      // Map thread IDs to worker-specific local maps for lockless updates
+      Map<Long, Map<ContainerKeyPrefix, Integer>> allLocalMaps = new ConcurrentHashMap<>();
+      
+      Object flushLock = new Object();
+      
+      Function<Table.KeyValue<String, OmKeyInfo>, Void> kvOperation = kv -> {
+        try {
+          // Get or create this worker's private local map using thread ID
+          Map<ContainerKeyPrefix, Integer> myLocalMap = allLocalMaps.computeIfAbsent(
+              Thread.currentThread().getId(), k -> new ConcurrentHashMap<>());
+          
+          handleKeyReprocess(kv.getKey(), kv.getValue(), myLocalMap, SHARED_CONTAINER_KEY_COUNT_MAP,
               reconContainerMetadataManager);
-          omKeyCount++;
 
-          // Check and flush data if it reaches the batch threshold
-          if (!checkAndCallFlushToDB(containerKeyMap, containerKeyFlushToDBMaxThreshold,
+          // Flush this worker's map when it reaches threshold
+          if (myLocalMap.size() >= PER_WORKER_THRESHOLD) {
+            synchronized (flushLock) {
+              if (!flushAndCommitContainerKeyInfoToDB(myLocalMap, Collections.emptyMap(),
+                  reconContainerMetadataManager)) {
+                throw new UncheckedIOException(new IOException("Unable to flush containerKey information to the DB"));
+              }
+            }
+          }
+          return null;
+        } catch (IOException e) {
+          throw new UncheckedIOException(e);
+        }
+      };
+      
+      try (ParallelTableIteratorOperation<String, OmKeyInfo> keyIter =
+               new ParallelTableIteratorOperation<>(omMetadataManager, omKeyInfoTable,
+                   StringCodec.get(), maxIterators, maxWorkers, maxKeysInMemory, PER_WORKER_THRESHOLD)) {
+        keyIter.performTaskOnTableVals(taskName, null, null, kvOperation);
+      }
+
+      // Final flush: Write remaining entries from all worker local maps to DB
+      for (Map<ContainerKeyPrefix, Integer> workerLocalMap : allLocalMaps.values()) {
+        if (!workerLocalMap.isEmpty()) {
+          if (!flushAndCommitContainerKeyInfoToDB(workerLocalMap, Collections.emptyMap(),
               reconContainerMetadataManager)) {
-            LOG.error("Failed to flush container key data for {}", taskName);
+            LOG.error("Failed to flush worker local map for {}", taskName);
             return false;
           }
         }
       }
 
-      // Final flush and commit
-      if (!flushAndCommitContainerKeyInfoToDB(containerKeyMap, containerKeyCountMap, reconContainerMetadataManager)) {
-        LOG.error("Failed to flush Container Key data to DB for {}", taskName);
+      // Capture total container count from shared map
+      long totalContainers = SHARED_CONTAINER_KEY_COUNT_MAP.size();
+
+      // Final flush: Shared container count map
+      if (!flushAndCommitContainerKeyInfoToDB(Collections.emptyMap(), SHARED_CONTAINER_KEY_COUNT_MAP, reconContainerMetadataManager)) {
```

Comment:
flush to be done at last, else can have twice flush.

Reply:
Implemented. Moved shared container count map flush inside the if (remainingTasks == 0) block. Now only the last completing task flushes the shared map once, preventing duplicate flushes.